tests/ld-floxlib.bats

```diff
@@ -0,0 +1,116 @@
+#     glibc version and confirm this does NOT match the pinned version
+#   - repeat with LD_AUDIT defined and confirm that the version again does
+#     not change
+#   - repeat with LD_LIBRARY_PATH=$FLOX_ENV_LIBS and confirm that the
+#     version does change
+# 2: confirm LD_AUDIT can find missing libraries
+#   - compile the get-nix-version program
+#   - observe that it can run the compiled program on the sample gif
+#   - unset LD_AUDIT and confirm it cannot run the program
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash
+
+# bats file_tags=end2end,ld-floxlib
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup
+}
+
+# ---------------------------------------------------------------------------- #
+
+# Helpers for project based tests.
+
+project_setup() {
+  export PROJECT_DIR="${BATS_TEST_TMPDIR?}/project-${BATS_TEST_NUMBER?}"
+  export PROJECT_NAME="${PROJECT_DIR##*/}"
+  rm -rf "$PROJECT_DIR"
+  mkdir -p "$PROJECT_DIR"
+  cp ./harnesses/ld-floxlib/* "$PROJECT_DIR"
+  pushd "$PROJECT_DIR" > /dev/null || return
+  _PKGDB_GA_REGISTRY_REF_OR_REV="${PKGDB_NIXPKGS_REV_OLDER?}" \
+    "$FLOX_BIN" init
+}
+
+project_teardown() {
+  popd > /dev/null || return
+  rm -rf "${PROJECT_DIR?}"
+  unset PROJECT_DIR
+  unset PROJECT_NAME
+}
+
+# ---------------------------------------------------------------------------- #
+
+setup() {
+  common_test_setup
+  project_setup
+}
+teardown() {
+  project_teardown
+  common_test_teardown
+}
+
+# ---------------------------------------------------------------------------- #
+#
+@test "test ld-floxlib.so on Linux only" {
+  if [ $(uname -s) != "Linux" ]; then
+    skip "not Linux"
+  fi
+
+  # Note:
+  # - installing old versions of nix (2.13.3) and glibc (2.35) for use in tests
+  # - installing curl and libarchive because those packages provide libraries
+  #   that are runtime dependencies of libnixmain.so
+  run env _PKGDB_GA_REGISTRY_REF_OR_REV="${PKGDB_NIXPKGS_REV_OLDER?}" \
+    "$FLOX_BIN" install curl gcc glibc libarchive nix patchelf
+  assert_success
+  assert_output --partial "✅ 'curl' installed to environment"
+  assert_output --partial "✅ 'gcc' installed to environment"
+  assert_output --partial "✅ 'glibc' installed to environment"
+  assert_output --partial "✅ 'libarchive' installed to environment"
+  assert_output --partial "✅ 'nix' installed to environment"
+  assert_output --partial "✅ 'patchelf' installed to environment"
+
+  ### Test 1: load libraries found in $FLOX_ENV_LIB_DIRS last
+  run "$FLOX_BIN" activate -- bash ./test-load-library-last.sh < /dev/null
+  assert_success
+
+  ### Test 2: confirm LD_AUDIT can find missing libraries
+  # Build print-nix-version, remove RUNPATH & interpreter
+  run "$FLOX_BIN" activate -- bash -exc '" \
+    g++ -o get-nix-version ./get-nix-version.cc -lnixmain && \
```

Comment:
question nonblocking: does `g++` need to be run inside the `activate`? What is it picking up from the activated environment?

Reply:
Yes it's the activated environment that sets LIBRARY_PATH and CPATH which is what g++ uses to find include files and libraries at compile time.

<pre><font color="#5F5FAF"><b>flox </b></font><font color="#FFAF87"><b>[tmp.vYoEM7vY0o]</b></font> <font color="#26A269">brantley@fedora</font>:<font color="#26A269">/tmp/tmp.vYoEM7vY0o</font>$ env | egrep &apos;(CPATH|LIBRARY_PATH)&apos;
<font color="#C01C28"><b>LIBRARY_PATH</b></font>=/tmp/tmp.vYoEM7vY0o/.flox/run/aarch64-linux.tmp.vYoEM7vY0o/lib
<font color="#C01C28"><b>CPATH</b></font>=/tmp/tmp.vYoEM7vY0o/.flox/run/aarch64-linux.tmp.vYoEM7vY0o/include
<font color="#5F5FAF"><b>flox </b></font><font color="#FFAF87"><b>[tmp.vYoEM7vY0o]</b></font> <font color="#26A269">brantley@fedora</font>:<font color="#26A269">/tmp/tmp.vYoEM7vY0o</font>$ 
</pre>